keras/applications/nasnet.py

```diff
@@ -167,7 +167,7 @@ def NASNet(input_shape=None,
                                       default_size=default_size,
                                       min_size=32,
                                       data_format=K.image_data_format(),
-                                      require_flatten=include_top or weights,
+                                      require_flatten=include_top and weights,
```

Comment:
This model uses global pooling layers rather than a `Flatten` layer, so this value should always be False.

Reply:
oh oh.
So the error message is still not complete yet, isn't it?

`ValueError: When setting`include_top=True` and loading `imagenet` weights, `input_shape` should be (331, 331, 3).`

thank you for your comment and I try to read a code more carefully.

Reply:
What happens if you remove this error and use the network with `include_top=True` and `weights='imagenet'` and a different input size?

Reply:
I want to use pre-trained model w/ small image, 32 <= weight = height < 331.

so I try this code;
```
from keras.applications.nasnet import NASNetLarge
model = NASNetLarge(weights='imagenet', include_top=False, input_shape=(100,100,3))
```
but, this raise error;
```
ValueError: When setting `include_top=True` and loading `imagenet` weights, `input_shape` should be (331, 331, 3).
```

this error called from `_obtain_input_shape` in `imagenet_utils.py`, and it checked a `nasnet.py`'s require_flatten, so I changed there to match the error message.

Reply:
Yes, but if remove this ValueError statement, are you able to build and use the network?

Reply:
In that case, the check and ValueError can safely be removed.